ui/src/components/Sidebar.tsx

```diff
@@ -1168,6 +1178,123 @@ function ExportButtons() {
   );
 }
 
+function TocPodButton({ id }) {
+  const store = useContext(RepoContext);
+  if (!store) throw new Error("Missing BearContext.Provider in the tree");
+  const setCursorNode = useStore(store, (state) => state.setCursorNode);
+
+  const codeMap = useStore(store, (state) => state.getCodeMap());
+  const wsRun = useStore(store, (state) => state.wsRun);
+
+  const [anchorEl, setAnchorEl] = useState(null);
+
+  const open = Boolean(anchorEl);
+
+  const handleClick = (event) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+  return (
+    <ListItem disablePadding>
+      <ListItemButton
+        id={id}
+        onClick={() => {
+          setCursorNode(id);
+        }}
+      >
+        <ListItemText primary={id} />
+      </ListItemButton>
+      <IconButton aria-label="More options" onClick={handleClick}>
+        <MoreVertIcon />
+      </IconButton>
+      <Popover open={open} anchorEl={anchorEl} onClose={handleClose}>
+        <MenuList>
+          <MenuItem>Rename</MenuItem>
+          <MenuItem>Copy</MenuItem>
+          {codeMap.has(id) ? (
+            <MenuItem
+              onClick={() => {
+                wsRun(id);
+                handleClose();
+              }}
+            >
+              Run
+            </MenuItem>
+          ) : null}
+          <MenuItem>Delete</MenuItem>
+        </MenuList>
+      </Popover>
+    </ListItem>
+  );
+}
+
+function dfs(
+  nodeId: string,
+  toc: Map<string, ReactflowNode[]>,
+  nodesMap: Y.Map<ReactflowNode<NodeData>>
+) {
+  let jsx = <></>;
+
+  if (toc.has(nodeId) && toc.get(nodeId)!.length > 0) {
+    toc.get(nodeId)?.forEach((child) => {
+      jsx = (
+        <>
+          {jsx}
+          <TocPodButton id={child.id} />
+          {toc.has(child.id) ? <ul>{dfs(child.id, toc, nodesMap)}</ul> : null}
```

Comment:
You are using `<ul>` to control padding and using `ListItem` to show lists. These two things sound like two alternatives to achieve the same functionality, not to be used together.

How about using MUI's [TreeView](https://mui.com/material-ui/react-tree-view/)? It seems to offer a good padded UI with additional features such as multi-selection.

Reply:
[TreeView](https://mui.com/material-ui/react-tree-view/) is indeed a better option.